debugger_ui: Update new process modal to include more context about its source

matt-revell opened the pull request:
Closes #36280

Release Notes:
  - Added additional context to debug task selection

Adding additional context when selecting a debug task to help with projects that have multiple config files with similar names for tasks. 

I think there is room for improvement, especially adding context for a LanguageTask type. I started but it looked like it would need to add a path value to that and wanted to make sure this was a good idea before working on that.

Also any thoughts on the wording if you do like this format? 

---

<img width="1246" height="696" alt="image" src="https://github.com/user-attachments/assets/b42e3f45-cfdb-4cb1-8a7a-3c37f33f5ee2" />

matt-revell committed:
Update debug modal to include more context about its source.

matt-revell committed:
Merge branch 'zed-industries:main' into feature/debug_context

osiewicz reviewed (dismissed):
A couple of thoughts:
1. We probably should not be dropping language icons altogether.
2. I'd love to have some consistency between worktree/global tasks. 
  - I think global tasks should just use an absolute path to the debug.json file
  - Worktree sources can use a relative path to the `debug.json` too, and not to the parent directory.

matt-revell committed:
Merge branch 'zed-industries:main' into feature/debug_context

matt-revell committed:
More consistant formatting for task file location.

Update the new process modal to have the path to the file containing the
task configuration to be under instead of next to the name.

matt-revell committed:
run clipply ...

matt-revell committed:
Use the full path for global config

matt-revell committed:
Merge branch 'zed-industries:main' into feature/debug_context

matt-revell committed:
Merge branch 'zed-industries:main' into feature/debug_context

matt-revell committed:
Merge branch 'zed-industries:main' into feature/debug_context

matt-revell committed:
Add file path to language tasks

matt-revell committed:
Merge branch 'zed-industries:main' into feature/debug_context

Anthony-Eid commented:
@matt-revell Thank you! I'll be able to review the code by the end of next week and most likely tomorrow or Tuesday

Anthony-Eid commented:
@matt-revell good job on this PR so far and thank you for your work! I'm going to go ahead and make so changes to get it mergable.

Anthony-Eid committed:
Refactor debug modal icon generation

We need extract the language name without getting rid of the task source
kind data from the debug picker candidate list. I did this so we could
use the task source kind data to generate each entries subtitle in the
picker (the path where the scenario came from)

This is a better solution because it relies on using a single source of
truth instead of getting the abs file path from the task context. Which
would be inaccurate when looking at files outside of worktree of origin
of debug scenarios

matt-revell committed:
Add the absolute path to a language task

If there is more than one workspace use the absolute path
instead of the relative path for a workspace debug task.

Anthony-Eid commented:
> I should have some time if you have not started working on the todo

I just saw your changes, thank you! This looks good to merge but I'm going to do one final review in the morning when I have more energy to double check everything

matt-revell committed:
Merge remote-tracking branch 'origin/main' into debug

matt-revell committed:
Merge branch 'zed-industries:main' into feature/debug_context

Anthony-Eid committed:
Add basic regression test and clean up code

Anthony-Eid committed:
Account for vscode paths too

Anthony-Eid committed:
Fix clippy error

Anthony-Eid committed:
Fix failing windows test
